src/Database/Query.php

```diff
@@ -2073,8 +2090,16 @@ public function selectTypeMap(TypeMap $typeMap = null)
      */
     protected function _decorateStatement($statement)
     {
+        $typeMap = $this->getSelectTypeMap();
+        $driver = $this->getConnection()->getDriver();
+
+        if ($this->typeCastEnabled && $typeMap->toArray()) {
+            $driver = $this->_connection->getDriver();
```

Comment:
Do you need to fetch the driver again? You already got it on line 2094.

Reply:
Yeah, bad copy paste. Thanks!